mlir/lib/Transforms/Utils/Utils.cpp

```diff
@@ -443,6 +460,16 @@ LogicalResult mlir::normalizeMemRef(AllocOp allocOp, Value *normalizedMemRef) {
   auto oldMemRef = allocOp.getResult();
   SmallVector<Value, 4> symbolOperands(allocOp.getSymbolicOperands());
 
+  // TODO(avarmapml) : Handle the initial filter to ensure that the only uses
+  //                   of memref is DeallocOp, CallOp, (maybe) ReturnOp
+  // if(llvm::any_of(oldMemRef.getUsers(),
+  //                 [](Operation *op) { 
+  //                   if(isMemRefDereferencingOp(op)) {
+  //                     return !(isa<DeallocOp>(op) || isa<CallOp>(op))
+  //                   }
+  //                 })) {
+  //   llvm::dbgs()<<"No going to support such"
+  // }
```

Comment:
You can use `#if 0` /  `#endif` instead of commenting each line. In any case, please drop commented out code - you can keep the TODO, but store the commented out code separately for your reference if needed and not in this PR.